File: lollys-lotto-ts-sdk/src/codegen/instructions/closeLottoGame.ts
import { TransactionInstruction, PublicKey, AccountMeta } from "@solana/web3.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import BN from "bn.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as borsh from "@coral-xyz/borsh" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as types from "../types" // eslint-disable-line @typescript-eslint/no-unused-vars
import { PROGRAM_ID } from "../programId"

export interface CloseLottoGameAccounts {
  authority: PublicKey
  lottoGame: PublicKey
  systemProgram: PublicKey
}

export function closeLottoGame(
  accounts: CloseLottoGameAccounts,
  programId: PublicKey = PROGRAM_ID
) {
  const keys: Array<AccountMeta> = [
    { pubkey: accounts.authority, isSigner: true, isWritable: true },
    { pubkey: accounts.lottoGame, isSigner: false, isWritable: true },
    { pubkey: accounts.systemProgram, isSigner: false, isWritable: false },
  ]
  const identifier = Buffer.from([73, 152, 130, 19, 160, 60, 59, 224])
  const data = identifier
  const ix = new TransactionInstruction({ keys, programId, data })
  return ix
}

File: lollys-lotto-rust-sdk/src/instructions/close_lotto_game.rs
use crate::instructions::*;

pub fn close_lotto_game(lotto_game: &Pubkey, authority: &Pubkey) -> Instruction {
    let data = lollys_lotto::instruction::CloseLottoGame.data();
    let accounts = lollys_lotto::accounts::CloseLottoGame {
        authority: *authority,
        lotto_game: *lotto_game,
        system_program: system_program::ID,
    }
    .to_account_metas(None);
    Instruction {
        program_id: lollys_lotto::ID,
        accounts,
        data,
    }
}
